Add retries to Write tasks page (#14601)
PrefectHQ/prefect: docs/3.0rc/develop/write-tasks.mdx

description: Learn the basics of writing tasks.
---

> A Prefect task is a Python function decorated with `@task` that represents a discrete unit of work in a Prefect workflow. 

Reviewer: ```suggestion
A Prefect task is a discrete unit of work in a Prefect workflow. 
You can turn any Python function into a task by adding an `@task` decorator to it.
```

## Example task

> Here's an example that moves a request from a flow into a task:

Reviewer: ```suggestion
Here's an example of a simple flow with a single task:
```

from prefect import task
import time

@task(timeout_seconds=1, log_prints=True)
def show_timeouts():
    print("I will execute")
    time.sleep(5)
    print("I will not execute")
```

## Retries

Prefect can automatically retry task runs on failure. 
A task run _fails_ if its Python function raises an exception.

To enable retries, pass `retries` and `retry_delay_seconds` arguments to your
task. 
If the task run fails, Prefect will retry it up to `retries` times, waiting
`retry_delay_seconds` seconds between each attempt. 
If the task fails on the final retry, Prefect marks the task as _failed_.

A new task run is not created when a task is retried. 
Instead, a new state is added to the state history of the original task run.

> Retries are often useful in cases that depend uon external systems, such as when making an API request. 

Reviewer: ```suggestion
Retries are often useful in cases that depend upon external systems, such as making an API request. 
```
